Documentation/stackhead-modules/development/module-configuration-file.md

```diff
@@ -18,6 +20,21 @@ terraform:
 ```
 {% endcode %}
 
+## constraints
+
+Constraints are used to ensure a StackHead module is compatible with other components.
+You can define which versions of StackHead the module is compatible with.
+
+The contraint format follows [semantic versioning](https://semver.org).
+
+```yaml
+---
+constraints:
+  - stackhead>=1.0.0 # All releases greater or equal than version 1.0.0
+  - stackhead ^1.0.0 # All minor and patch releases (>=1.0.0 <2.0.0)
+  - stackhead ~1.0.0 # All patch releases (>=1.0.0 <1.1.0)
+```
+
```

Comment:
Add note:

We recommend to allow any release after the version that is known to work (e.g. `stackhead>=1.0.0`). Once a version is known to be incompatible, the constraint can be adjusted. This is to make sure modules do not have to be updated once a new (breaking) StackHead is published.